modules/calib3d/src/fundam.cpp

```diff
@@ -377,7 +377,8 @@ cv::Mat cv::findHomography( InputArray _points1, InputArray _points2,
         p.reshape(2, npoints).convertTo(m, CV_32F);
     }
 
-    CV_Assert( src.checkVector(2) == dst.checkVector(2) );
+    // Need atleast 4 point correspondences
+    CV_Assert( src.checkVector(2) == dst.checkVector(2) && src.checkVector(2) > 3 );
```

Comment:
1. Cache expression result: `src.checkVector(2)` is used twice - no need to evaluate it twice.
2. Avoid expanding of assertion check. Such error messages are not user friendly. Use one check per assertion or better to use `CV_Check*()` macros.
3. Please add simple test for this case.

Reply:
I've updated the commit with a better check here